run_tests.py

```diff
@@ -80,24 +83,47 @@ class GRREverythingTestLoader(test_lib.GRRTestLoader):
   base_class = test_lib.GRRBaseTest
 
 
+class GRROneTestLoader(test_lib.GRRTestLoader):
+  """Load one specific GRR test case."""
+  base_class = test_lib.GRRBaseTest
+
+  def loadTestsFromModule(self, _):
+    """Just return all the tests as if they were in the same module."""
+    test_cases = [
+        self.loadTestsFromTestCase(x) for x in self.base_class.classes.values()
+        if x.__name__ == self.test_class_name]
+
+    return self.suiteClass(test_cases)
+
+  def __init__(self, test_class_name):
+    self.test_class_name = test_class_name
+
+    super(GRROneTestLoader, self).__init__()
+
+
 def RunTest(test_suite, stream=None):
   out_fd = stream
   if stream:
     out_fd = StringIO.StringIO()
 
   try:
-    test_lib.GrrTestProgram(argv=[sys.argv[0], test_suite],
-                            testLoader=GRREverythingTestLoader(
-                                labels=flags.FLAGS.labels),
-                            testRunner=unittest.TextTestRunner(
-                                stream=out_fd))
+    program = test_lib.GrrTestProgram(
+                argv=[sys.argv[0], test_suite],
+                testLoader=GRROneTestLoader(
+                    test_class_name=test_suite),
+                testRunner=unittest.TextTestRunner(
+                    stream=out_fd))
+  except SystemExit as e:
+    # must include this section, as GrrTestProgram issues
```

Comment:
Regarding your comment about the GrrProgram bug: Yes, it is a bug.

In travis, they don't run in their own process, because I'm specifying `--processes=1` in the invocation, so it never runs in multiprocessing mode: It runs in single processing mode, and all the tests run sequentially in a single process. This is intentional, as otherwise we run out of memory in travis, so we must run processes sequentially. There's also additional limits in travis, but they're not all clear to me. For example, I suspect there's a limit in open files and a limit of running processes that may or may not be causing problems. [See this failing build as an example](https://s3.amazonaws.com/archive.travis-ci.org/jobs/45939849/log.txt)

The bug is not that the GrrTestProgram issues a `sys.exit` – this is normal, as it extends the unittest [TestProgram](https://pythonhosted.org/gchecky/unittest.TestProgram-class.html), which is supposed to be a stand-alone CLI program that runs tests. It's not meant to be called from within functions like we are doing here, so this code was and still is wrong. This is what the TestRunner is for. All tests are behaving like this. It's not a bug in a specific test. It's a bug in the fact that a unit test program is invoked within a function, a misuse of the Python unit testing framework.

In fact, automated test discovery and running frameworks such as [nose](https://nose.readthedocs.org/en/latest/) provide for better such mechanisms and we should not reinventing the wheel with buggy behavior or incorrect use of the provided framework. The only reason why nose is not easily pluggable to the current implementation is our use of Google flags, but I believe this can be worked around.

So, for now I recommend that we don't require the whole testing framework to be rewritten in this one pull request to be correct. This is a pull request that improves what we currently have and works around some of the bugs, and fixes others, while it doesn't introduce new ones. In future pull requests I'll try to tidy up the code more and to properly fix this wrong invocation. This should still be merged as it is.